crates/sage/src/endpoints/data.rs

```diff
@@ -711,132 +697,106 @@ impl Sage {
         })
     }
 
-    async fn transaction_coin(&self, db: &Database, coin: CoinStateRow) -> Result<TransactionCoin> {
-        let coin_id = coin.coin_state.coin.coin_id();
-
-        let (kind, p2_puzzle_hash) = match coin.kind {
-            CoinKind::Unknown => (AssetKind::Unknown, None),
-            CoinKind::Xch => (AssetKind::Xch, Some(coin.coin_state.coin.puzzle_hash)),
+    async fn transaction_coin(&self, transaction_coin: SqliteRow) -> Result<TransactionCoin> {
+        let coin_id: Option<Bytes32> = to_bytes32_opt(transaction_coin.get("coin_id"));
+        let kind_int: i64 = transaction_coin.get("kind");
+        let coin_kind = CoinKind::from_i64(kind_int);
+        let p2_puzzle_hash: Option<Bytes32> =
+            to_bytes32_opt(transaction_coin.get("p2_puzzle_hash"));
+        let name: Option<String> = transaction_coin.get("name");
+        let item_id: Option<Bytes32> = to_bytes32_opt(transaction_coin.get("item_id"));
+        let amount: Vec<u8> = transaction_coin.get("amount");
+
+        let kind = match coin_kind {
+            CoinKind::Unknown => AssetKind::Unknown,
+            CoinKind::Xch => AssetKind::Xch,
             CoinKind::Cat => {
-                if let Some(cat) = db.cat_coin(coin_id).await? {
-                    if let Some(row) = db.cat(cat.asset_id).await? {
-                        (
-                            AssetKind::Cat {
-                                asset_id: hex::encode(cat.asset_id),
-                                name: row.name,
-                                ticker: row.ticker,
-                                icon_url: row.icon,
-                            },
-                            Some(cat.p2_puzzle_hash),
-                        )
-                    } else {
-                        (
-                            AssetKind::Cat {
-                                asset_id: hex::encode(cat.asset_id),
-                                name: None,
-                                ticker: None,
-                                icon_url: None,
-                            },
-                            Some(cat.p2_puzzle_hash),
-                        )
+                if let Some(item_id) = item_id {
+                    AssetKind::Cat {
+                        asset_id: hex::encode(item_id),
+                        name,
+                        ticker: transaction_coin.get("ticker"),
+                        icon_url: transaction_coin.get("icon"),
                     }
                 } else {
-                    (AssetKind::Unknown, None)
+                    AssetKind::Unknown
                 }
             }
             CoinKind::Nft => {
-                if let Some(nft) = db.nft_by_coin_id(coin_id).await? {
-                    let row = db.nft_row(nft.info.launcher_id).await?;
-
-                    let mut allocator = Allocator::new();
```

Comment:
@Rigidity is this still the standard way of getting the NFT icon? I haven't ported this final piece over quite yet.